components/camel-zipfile/src/main/java/org/apache/camel/dataformat/zipfile/ZipFileDataFormat.java

```diff
@@ -72,6 +73,7 @@ public void marshal(final Exchange exchange, final Object graph, final OutputStr
 
     @Override
     public Object unmarshal(final Exchange exchange, final InputStream inputStream) throws Exception {
+        exchange.getIn().setHeader("unzipEmptyDirectorySupported", isSupportEmptyDirectory());
         if (usingIterator) {
             return new ZipIterator(exchange.getIn());
```

Comment:
Do not use a Camel message header for this, but set the option on the iterator using getter/setter

    ZipIterator it = new ...
    it.setUnzipEmptyDirectory(...)